src/OrleansTestingHost/TestCluster.cs

```diff
@@ -107,7 +110,7 @@ public void Deploy(IEnumerable<string> siloNames)
         /// Deploys the cluster using the specified configuration and starts the client in-process.
         /// </summary>
         /// <param name="siloNames">Only deploy the specified silos which must also be present in the <see cref="Runtime.Configuration.ClusterConfiguration.Overrides"/> collection.</param>
-        public async Task DeployAsync(IEnumerable<string> siloNames)
+        public async Task DeployAsync(IEnumerable<string> siloNames, IServiceProvider serviceProvider)
```

Comment:
yes agree here, that's why I have the TODO keyword in Silo constructor on "need to inject real depdendency instead of the whole ServiceProvider"